test/compilable/bug21196.d

```diff
@@ -1,5 +1,5 @@
 /*
-REQUIRED_ARGS: -de
+REQUIRED_ARGS: -de -revert=dip25
```

Comment:
this seems unrelated

Reply:
Nope. Otherwise this fails.

Reply:
Then this isn't a refactoring.

Reply:
I should explain better. The code before (which made this test pass) was:

```d
             if (emitError)
                    error(e.loc, msg, e.toChars(), kind, v.toChars());
                else if (!sc.isDeprecated())
                    deprecation(e.loc, msg, e.toChars(), kind, v.toChars());
```

Notice that the deprecation was only printed in the case of `!sc.isDeprecated`. That was an ad-hoc solution to https://issues.dlang.org/show_bug.cgi?id=21196 whereas the real way to get rid of deprecation warnings like the ones here is to pass `-revert=previewXXX`. That's why `noop` exists now.

Reply:
Hold on, does it mean that with this change, deprecations will show up in deprecated scopes ?

Reply:
> Hold on, does it mean that with this change, deprecations will show up in deprecated scopes ?

Example?

If you mean that, in the original issue you filed, that they'll show up, then yes, *unless* `-revert=preview25` is used.

Reply:
Then it's introducing a regression. Having deprecations, *even language deprecations*, be silenced when in a `deprecated` scope is an important feature, as it allows libraries to transition away from one API to the other.
The use case that prompted the issue was a library that would have an overload which internally kept pointers to stack allocated data in `void[]` slices. Transitioning to the other, safer overload was trivial, so in the library we deprecated the function that relied on the non-dip25 behavior. The other alternative would be either have everyone see an unrelated deprecation showing up on each build, or force us to make a breaking change outside of our release cycle, none of which is acceptable.

Reply:
> Example?

This very test case. Both functions are deprecated and aren't called from a deprecated scope, hence no deprecation should be issued.

Reply:
I clearly didn't understand that issue or the test for it. Fixing it now.